source/src/cip/cipqos.c

```diff
@@ -88,38 +106,51 @@ EipStatus SetAttributeSingleQoS(
   return kEipStatusOkSend;
 }
 
-CipUsint GetPriorityForSocket(ConnectionObjectPriority priority) {
+CipUsint CipQosGetDscpPriority(ConnectionObjectPriority priority) {
+
+  CipUsint priority_value;
+
+  if (!s_is_frozen) {
+    s_is_frozen = true;
+    s_frozen.dscp_event = dscp_event;
+    s_frozen.dscp_general = dscp_general;
+    s_frozen.dscp_urgent = dscp_urgent;
+    s_frozen.dscp_scheduled = dscp_scheduled;
+    s_frozen.dscp_high = dscp_high;
+    s_frozen.dscp_low = dscp_low;
+    s_frozen.dscp_explicit = dscp_explicit;
+  }
 
-  CipUsint priority_value = dscp_explicit;
+  priority_value = s_frozen.dscp_explicit;
   switch (priority) {
     case kConnectionObjectPriorityLow:
-      priority_value = dscp_low;
+      priority_value = s_frozen.dscp_low;
       break;
     case kConnectionObjectPriorityHigh:
-      priority_value = dscp_high;
+      priority_value = s_frozen.dscp_high;
       break;
     case kConnectionObjectPriorityScheduled:
-      priority_value = dscp_scheduled;
+      priority_value = s_frozen.dscp_scheduled;
       break;
     case kConnectionObjectPriorityUrgent:
-      priority_value = dscp_urgent;
+      priority_value = s_frozen.dscp_urgent;
       break;
     default:
```

Comment:
update to also have kConnectionObjectPriorityExplicit

Reply:
To have another case label with kConnectionObjectPriorityExplicit may be nice but the "enum {} kConnectionObjectPriority" from cipconnectionobject.h doesn't have such a member. I didn't dare to add such an enum constant because I'm not familiar with the internals of that module.

The lack of kConnectionObjectPriorityExplicit also has its impact on the code with  CipQosGetDscpPriority(0xFFFF) in generic_networkhandler.c.

So if there are no side effects from adding kConnectionObjectPriorityExplicit to the mentioned enum then this change should be done.

Reply:
Yeah. that was a wrong decision made by me, so I didn't expect you to know or fix that. This comment is more a reminder for myself to change that when I am on it. Otherwise, this might get lost